app/src/main/java/ch/epfl/sweng/SDP/home/leaderboard/Player.java

```diff
@@ -0,0 +1,127 @@
+                   boolean isCurrentUser) {
+        this.context = context;
+        this.userId = userId;
+        this.username = username;
+        this.trophies = trophies;
+        this.league = league;
+        this.isCurrentUser = isCurrentUser;
+    }
+
+    String getUserId() {
+        return userId;
+    }
+
+    /**
+     * Allows one to call sort on a collection of Players. Sorts collection in descending
+     * order.
+     *
+     * @param object Player to compare
+     * @return 1 if this is larger, -1 if this is smaller, 0 else
+     */
+    @Override
+    public int compareTo(Object object) {
+        int compareTrophies = -this.trophies.compareTo(((Player) object).trophies);
+        if (compareTrophies == 0) {
+            return this.username.compareTo(((Player) object).username);
+        }
+        return compareTrophies;
+    }
+
+    boolean playerNameContainsString(String query) {
+        return username.toUpperCase().contains(query.toUpperCase());
+    }
+
+    /**
+     * Converts this player into a LinearLayout that will be displayed on the leaderboard.
+     *
+     * @param index   of the player
+     * @return LinearLayout that will be displayed
+     */
+    @SuppressLint("NewApi")
+    LinearLayout toLayout(int index) {
+        TextView usernameView = new TextView(context);
+        Resources res = context.getResources();
+        styleView(usernameView, rank + ". " + username, res.getColor(
+                isCurrentUser ? R.color.colorPrimaryDark : R.color.colorDrawYellow),
+                new LinearLayout.LayoutParams(0, LinearLayout.LayoutParams.WRAP_CONTENT, 4));
+        usernameView.setPadding(0, 10, 0, 10);
+
+        TextView trophiesView = new TextView(context);
+        styleView(trophiesView, trophies.toString(),
+                res.getColor(R.color.colorPrimaryDark),
+                new LinearLayout.LayoutParams(0,
+                        LinearLayout.LayoutParams.WRAP_CONTENT, 2));
+
+        trophiesView.setTextAlignment(RelativeLayout.TEXT_ALIGNMENT_TEXT_END);
+
+        ImageView leagueView = new ImageView(context);
+        leagueView.setLayoutParams(new LinearLayout.LayoutParams(0,
+                LinearLayout.LayoutParams.MATCH_PARENT, 1));
+        leagueView.setImageResource(getLeagueImageId(league));
+
+        final FriendsButton friendsButton =
+                new FriendsButton(context, this, index, isCurrentUser);
+
+        LinearLayout entry = addViews(new LinearLayout(context),
+                new View[]{usernameView, trophiesView, leagueView, friendsButton});
+
+        entry.setBackgroundColor(res.getColor(
+                isCurrentUser ? R.color.colorDrawYellow : R.color.colorLightGrey));
+        entry.setPadding(30, 10, 30, 10);
+
+        return entry;
+    }
+
+    private LinearLayout addViews(LinearLayout layout, View[] views) {
+        for (int i = 0; i < views.length; ++i) {
```

Comment:
Here you can use a for-each

Reply:
I did already try, but forEach was only introduced in java8 i think?

Reply:
I think it was introduced in Java 5! It doesn't work?

Reply:
Ah I see which forEach you mean. You're right.